### ISSUE TITLE
[Bug]: _fwd_kernel_ep_scatter_2 ignores recv_x_scale_stride1 for column-major f32 scales when num_tokens >= 2

### ISSUE THREAD

**Issue by Suppressor72** [vllm-project/vllm#53338]
## Environment

- vLLM nightly `0.26.1rc1.dev926+gb05ae5dc0` (collect_env output attached below
  attached below); bug present in current `main` source of
  `deep_gemm_utils.py`.
- Hardware: 2× RTX 5090 (SM120), TP=2, CUDA 12.x.
- Model: Qwen3.6-35B-A3B-FP8 (DeepGEMM grouped-FP8 contiguous MoE path).

<details>
<summary>collect_env output</summary>

```
Collecting environment information...
uv is set
==============================
        System Info
==============================
OS                           : Ubuntu 26.04 LTS (x86_64)
GCC version                  : (Ubuntu 15.2.0-16ubuntu1) 15.2.0
Clang version                : Could not collect
CMake version                : version 3.31.6
Libc version                 : glibc-2.43

==============================
       PyTorch Info
==============================
PyTorch version              : 2.13.0+cu130
Is debug build               : False
CUDA used to build PyTorch   : 13.0
ROCM used to build PyTorch   : N/A
XPU used to build PyTorch    : N/A

==============================
      Python Environment
==============================
Python version               : 3.12.13 (main, Jul 18 2026, 17:02:19) [Clang 22.1.3 ] (64-bit runtime)
Python platform              : Linux-7.0.0-29-generic-x86_64-with-glibc2.43
    
==============================
       CUDA / GPU Info
==============================
Is CUDA available            : True
CUDA runtime version         : Could not collect
CUDA_MODULE_LOADING set to   : 
GPU models and configuration : 
GPU 0: NVIDIA GeForce RTX 5090
GPU 1: NVIDIA GeForce RTX 5090

Nvidia driver version        : Could not collect
cuDNN version                : Could not collect
HIP runtime version          : N/A
MIOpen runtime version       : N/A

```

</details>

## Summary

In the non-`PACK_UE8M0` (f32-scales) branch of
`_fwd_kernel_ep_scatter_2`
(`vllm/model_executor/layers/fused_moe/deep_gemm_utils.py`), the scale
row is loaded as:

```python
to_copy_s = tl.load(
    recv_x_scale + token_id * recv_x_scale_stride0 + offset_in_s,
    mask=mask_s,
)
```

`offset_in_s` (the group index) is added **without multiplying by
`recv_x_scale_stride1`**, although the kernel receives that stride and
uses it correctly in the `PACK_UE8M0` branch (`g0 *
recv_x_scale_stride1`, …). The f32 branch therefore assumes a
row-major scale tensor (`stride1 == 1`).

## Impact

`per_token_group_quant_fp8(..., column_major_scales=True)` — the
MoE-input quant used on this path — returns scales with strides
`(1, num_tokens)`. At `num_tokens == 1` the layout coincides with
row-major and the read is accidentally correct. At `num_tokens >= 2`
the kernel reads address `token_id + g` instead of
`token_id + g * num_tokens`, so each scattered row receives scales
from the wrong token/group coordinates — other tokens' group scales
are copied into the expert-workspace scale rows that the grouped GEMM
then consumes. In our random synthetic fixtures the mis-association
was numerically masked (max end-output delta ~1e-9, because adjacent
groups of a token frequently share a power-of-two scale), which makes
the bug easy to miss in tolerance-based tests; deterministic
constructions with distinct per-token magnitudes expose it
immediately.

## Reproducible example

```python
import torch
from vllm.model_executor.layers.quantization.utils.fp8_utils import (
    per_token_group_quant_fp8)
from vllm.model_executor.layers.fused_moe.deep_gemm_utils import (
    deepgemm_moe_permute)

c, H, E, topk = 2, 2048, 256, 8
dev = "cuda"
x = torch.zeros(c, H, dtype=torch.bfloat16, device=dev)
# token t, group g -> distinct per-token magnitude bands so every
# (t, g) has a unique power-of-two exponent
for t in range(c):
    for g in range(16):
        x[t, g * 128:(g + 1) * 128] = 2.0 ** (g - 8 + 20 * t)
topk_ids = torch.arange(c * topk, dtype=torch.int32,
                        device=dev).reshape(c, topk) % E

xq, xs = per_token_group_quant_fp8(
    x, 128, column_major_scales=True, use_ue8m0=True)
aq_out, aq_scale_out, m_indices, inv_perm, _ = deepgemm_moe_permute(
    aq=xq, aq_scale=xs, topk_ids=topk_ids, local_num_experts=E,
    expert_map=None, expert_tokens_meta=None)

# expected: the scale row scattered to each routed (t, k) destination
# equals the source token's quant scales
for t in range(c):
    for k in range(topk):
        dest = int(inv_perm[t, k])
        got = aq_scale_out[dest]          # f32, [M_sum, 16]
        exp = xs[t]                        # logical [16], col-major
        if not torch.equal(got.cpu(), exp.float().cpu()):
            print(f"MISASSOCIATED at t={t} k={k}: "
                  f"got exponents {torch.log2(got).tolist()} != "
                  f"expected {torch.log2(exp.float()).tolist()}")
            raise SystemExit(1)
print("all scale rows correct (bug not reproduced on this build)")
```

Observed (nightly above, `c=2`): every routed row fails — e.g. for a
source token whose true group-7 exponent is the odd one out, the
scattered row carries it at group 14, exactly the `t + g` addressing
(with strides `(1, 2)`, `t + g` for `g=7` aliases `t=1, g_true=7`
storage position of `t + 2*7`). Expected: bitwise-equal rows as the
assertion demands. At `c=1` the script passes (layout coincidence).

## Suggested fix

Mirror the packed branch — multiply the group offset by the stride:

```python
to_copy_s = tl.load(
    recv_x_scale + token_id * recv_x_scale_stride0
    + offset_in_s * recv_x_scale_stride1,
    mask=mask_s)
```

No signature change (the stride is already an argument). Regression
test: the script above at `num_tokens in (1, 2, 4)`.

Found while developing a standalone fused-prepare prototype for this
path. Happy to send a PR with the fix + regression test.

---

*Drafted with AI assistance and reviewed by the reporter; the
diagnosis and reported hardware observations were verified by the
reporter.*
